store/sqlstore/channel_store.go

```diff
@@ -30,6 +30,8 @@ const (
 	CHANNEL_MEMBERS_COUNTS_CACHE_SEC  = 1800 // 30 mins
 
 	CHANNEL_CACHE_SEC = 900 // 15 mins
+
+	TOTAL_ROLE_COUNT = 11
```

Comment:
Why 11?

Reply:
That's the [number of roles](https://github.com/mattermost/mattermost-server/pull/8232/files#diff-6a93e94fc8adb70b1a7468d9e2324560R172) for Phase 1 permissions. I had considered invoking `MakeDefaultRoles()` and then counting the result but it seemed like overkill to do each query, but I'm not totally opposed to it.

Phase 3 will introduce custom roles and that's when this count would need to be a query.

Reply:
Ok cool, if it's going to be fixed later anyways that's fine